### PR TITLE
WIP Parts of a start to narrative tutorial documentation

### PR DESCRIPTION
**Pull request by jasongrout**
Based on initial drafts at https://jupyter.hackpad.com/JupyterLab-Architecture-CTx9FgioXIZ

### CODE REVIEW

File: tutorial/repo.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# General Codebase Orientation
+
+The repository contains an npm package and a python package, both named `jupyterlab`.
+
+## Directories
+
+### NPM package: `src/`, `lib/`, `typings/`, `scripts/`
+
+* `src/`: the source Typescript files. `npm run build` builds the source files into javascript files in `lib/`. `npm run clean` deletes the `lib/` directory.
```
Comment (blink1073) [inline tutorial/repo.md]: `TypeScript` and `JavaScript`

**Comment by blink1073** [comment]
Overall it looks good to me.  After today I'll merge this with `docs/` and move the API to `docs/api`.

File: tutorial/repo.md (hunk 1 above)
Comment (blink1073) [inline tutorial/repo.md]: Should also use `Python` for consistency.

**Comment by jasongrout** [comment]
Great, thanks. I'm working on the notebook section, then I'll make a pass back through to fix things.

**Comment by jasongrout** [comment]
If you want to put your stuff in different files in this directory, then there won't be a merge conflict with us both working in two different PRs.

**Comment by blink1073** [comment]
Merging to iterate.

**Pull request merged**